ulib/experimental/Steel.Memory.fsti

```diff
@@ -308,19 +317,31 @@ effect MstTot
   (expects:slprop u#1)
   (provides: a -> slprop u#1)
   (frame:slprop u#1)
+  (pre:mprop expects)
+  (post:mprop2 expects provides)
   = NMSTTotal.NMSTATETOT a (full_mem u#1) mem_evolves
     (requires fun m0 ->
         inames_ok except m0 /\
-        interp (expects `star` frame `star` locks_invariant except m0) m0)
+        interp (expects `star` frame `star` locks_invariant except m0) m0 /\
+        pre (core_mem m0))
     (ensures fun m0 x m1 ->
         inames_ok except m1 /\
         interp (expects `star` frame `star` locks_invariant except m0) m0 /\  //TODO: fix the effect so as not to repeat this
         interp (provides x `star` frame `star` locks_invariant except m1) m1 /\
+        post (core_mem m0) x (core_mem m1) /\
```

Comment:
Yes, this seems pretty natural

Reply:
A nit is that I think the name MstTot was initially chosen as an analog of Tot, i.e., an effect without a pre and a post. It doesn't matter much right now, but we might consider a better name for this.

Reply:
That's a good point. A similar comment applies to the MstTot in Semantics